src/esp/assets/ResourceManager.cpp

```diff
@@ -1297,22 +1299,32 @@ void ResourceManager::addObjectToDrawables(const std::string& objTemplateHandle,
     // need a new node for scaling because motion state will override scale
     // set at the physical node
     scene::SceneNode& scalingNode = parent->createChild();
+    newNodes.push_back(&scalingNode);
     Magnum::Vector3 objectScaling = physicsObjectAttributes->getScale();
     scalingNode.setScaling(objectScaling);
 
-    addComponent(loadedAssetData.meshMetaData, scalingNode, lightSetup,
-                 drawables, loadedAssetData.meshMetaData.root);
+    std::vector<scene::SceneNode*> componentNodes =
+        addComponent(loadedAssetData.meshMetaData, scalingNode, lightSetup,
+                     drawables, loadedAssetData.meshMetaData.root);
+
+    for (auto node : componentNodes) {
+      newNodes.push_back(node);
+    }
   }  // should always be specified, otherwise won't do anything
+  return newNodes;
```

Comment:
How about something like this? 
```suggestion
    componentNodes.push_back(&scalingNode);
    newNodes = componentNodes;
    }
  }  // should always be specified, otherwise won't do anything
  return newNodes;
```

Reply:
🤔  This will break the depth-first ordering of the nodes. I don't think this matters now, just throwing it out there.

Reply:
Oh, I was not aware we were hoping to retain ordering.  Then I would still suggest we do the insert method like i have below, since it is more efficient.

Reply:
Agreed. I don't think there is a reason to maintain order, but we may as well if it's easy. Nice property to have 🤷‍♂️